docs/content/en/usage.md

```diff
@@ -10,28 +10,88 @@ This module globally injects a `$sanity` helper, meaning that you can access it
 
 ## Reference
 
-### `fetch`
+### fetch
 
-This enables you to perform a GROQ query against your Sanity dataset. By default it returns a `Promise<unknown>` although you can customise the type of the return.
+This enables you to perform a GROQ query against your Sanity dataset.
+
+#### Example with `asyncData`
+
+<code-group>
+  <code-block label="JavaScript" active>
+
+```js
+import { groq } from '@nuxtjs/sanity'
+
+const query = groq`{ "articles": *[_type == "article"]}`
+
+export default {
+  asyncData({ $sanity }) {
+    return $sanity.fetch(query)
+  },
+}
+```
+
+  </code-block>
+  <code-block label="TypeScript">
 
 ```ts
 import { groq } from '@nuxtjs/sanity'
 
+const query = groq`{ "articles": *[_type == "article"]}`
+
+export default {
+  asyncData({ $sanity }) {
+    // By default it returns a `Promise<unknown>`,
+    // but you can customise the type of the return.
+    return $sanity.fetch<string>(query)
```

Comment:
Probably should type this as something other than string given the query

Reply:
ah I see. I just copied the code from the existing, but actually in this case we get an object, not a string. I don't know much about TypeScript (planning on learning it for 2021, btw if you know any good free/paid resources, let me know 😉)
Here would it be something like this?
`return $sanity.fetch<{ articles: [] }>(query)`